cores/esp8266/core_esp8266_i2s.cpp

```diff
@@ -313,7 +313,7 @@ static bool _i2s_write_sample(uint32_t sample, bool nb) {
         if (tx->slc_queue_len > 0) {
           break;
         } else {
-          optimistic_yield(10000);
+          optimistic_yield(1000);
```

Comment:
Expectation wait time for a new buffer to be freed up by the HW DMA is (64/2) 32-bit samples at 44.1khz, so ~726us.

So the prior yield would very rarely occur, but this new one has a very high chance of hitting especially if the app has done other stuff before I2S write.

Because I2S is time critical (you underflow, you click and pop) I think increase the number of unpredictable calls to SYS context could increase issues and would not change the timeout here or in the other sections of I2S.